### PR TITLE
fix(Host): RHICOMPL-1415 os version search

### PR DESCRIPTION
**Pull request by akofink**
Signed-off-by: Andrew Kofink <akofink@redhat.com>

## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [X] Input Validation
- [X] Output Encoding
- [X] Authentication and Password Management
- [X] Session Management
- [X] Access Control
- [X] Cryptographic Practices
- [X] Error Handling and Logging
- [X] Data Protection
- [X] Communication Security
- [X] System Configuration
- [X] Database Security
- [X] File Management
- [X] Memory Management
- [X] General Coding Practices

### CODE REVIEW

File: app/models/concerns/host_searching.rb [hunk 1]
```diff
@@ -5,8 +5,12 @@ module HostSearching
   extend ActiveSupport::Concern
 
   included do
-    scoped_search on: %i[display_name os_major_version os_minor_version]
+    scoped_search on: %i[id display_name], only_explicit: true
     scoped_search on: :display_name, rename: :name
```
Comment (akofink) [inline app/models/concerns/host_searching.rb:9]: Everything is now `only_explicit: true` except name so we don't run into this again.

File: db/cyndi_setup_test.sql [hunk 2]
```diff
@@ -36,5 +36,5 @@ FROM inventory.hosts_v1_1;
 --
 -- These are the host seeds for use with testing locally and in CI
 --
-INSERT INTO inventory.hosts_v1_1 VALUES ('5e65ac3f-8b44-4f60-9e99-abdafb31740c', '00001', 'MyStringone', '[{"namespace": "insights-client", "key": "env", "value": "prod"}]', '2020-07-21 05:35:53.682554+00', '2020-07-21 05:35:53.682554+00', '2030-07-21 05:35:53.682554+00', '{"sap_system": true}');
+INSERT INTO inventory.hosts_v1_1 VALUES ('5e65ac3f-8b44-4f60-9e99-abdafb31740c', '00001', 'MyStringone', '[{"namespace": "insights-client", "key": "env", "value": "prod"}]', '2020-07-21 05:35:53.682554+00', '2020-07-21 05:35:53.682554+00', '2030-07-21 05:35:53.682554+00', '{"sap_system": true, "operating_system": {"major": 7, "minor": 4, "name": "RHEL"}}');
```
Comment (akofink) [inline db/cyndi_setup_test.sql:39]: In order to test this, I had to add some OS info here. Only one side effect was produced in the `xccdf/test_result_test.rb` below, which was fixed with a stub.

File: spec/api/v1/schemas/hosts.rb [hunk 3]
```diff
@@ -21,13 +21,13 @@ module Hosts
...
-              example: '7',
+              type: 'integer',
+              example: 7,
               nullable: true
             },
             os_minor_version: {
-              type: 'string',
-              example: '3',
+              type: 'integer',
+              example: 3,
```
Comment (akofink) [inline spec/api/v1/schemas/hosts.rb:30]: I decided to go ahead and update the type here to match inventory. I could revert it if you think it's a significant enough API breakage, but it seems pretty innocuous to me.

File: swagger/v1/openapi.json [hunk 4]
```diff
@@ -2541,8 +2541,8 @@
                     "type": "host",
                     "attributes": {
                       "name": "MyStringone",
-                      "os_major_version": null,
-                      "os_minor_version": null,
+                      "os_major_version": 7,
+                      "os_minor_version": 4,
```
Comment (akofink) [inline swagger/v1/openapi.json:2545]: Due to updating the cyndi_setup_test.sql

**Review by bastilian — APPROVED** [review]
LGTM! Thank you @akofink!

**Pull request merged**